### PR TITLE
feat: add file.mkdir and file.delete functions

### LINKED ISSUE #1267: Extend file module functions (for smoother GitOps)
Hello there!

KCL being a very capable versatile language should have more file system control functions:
  - mkdir
  - delete
 And maybe even more than this.
 
It's essential to have control over transitional file creation/deletion and placement. For instance in my case, I do generate prometheus rules from parameterized KCL definitions as the first step:

```
cd appsops/rules/graph && mkdir -p rules manifests && kcl -dY ./kcl.yaml
generate_rules:
- graph/rules/common-certmanager.yaml
- graph/rules/common-elasticsearch.yaml
- graph/rules/common-istio.yaml
- graph/rules/common-k8s-events.yaml
- graph/rules/common-k8s-nodes.yaml
- graph/rules/common-k8s-resources.yaml
- graph/rules/common-vector.yaml
```

This is the configuration entry for the next stage, which proceeds with VMRule manifests generation. Simply there should be enough control in place on where to place the files, how to create or delete them to _enable smooth GitOps experience_ out of box. In fact KCL can be and should be used independently to provide a solid GitOps experience.

### PR DESCRIPTION
**Pull request by shruti2522**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 
fix #1267 

#### 2. What is the scope of this PR (e.g. component or file name):

kcl-lang/kcl/kclvm/runner/src/linker.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

Added `kclvm_mkdir` and `kclvm_delete` module functions.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [x] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

**Comment by Peefy** [comment]
A reference PR to add more functions for the file module. https://github.com/kcl-lang/kcl/pull/1157

**Comment by shruti2522** [comment]
Ah, got it now. While going through the issue, I supposed them to be cli functions, that's why I made changes to the cli.

**Comment by shruti2522** [comment]
Ready for review @Peefy

**Comment by Peefy** [comment]
Could you please add more tests and code for the resolver like this PR https://github.com/kcl-lang/kcl/pull/1157

**Comment by shruti2522** [comment]
Sure

**Comment by shruti2522** [comment]
added the tests @Peefy, please have a look

File: kclvm/sema/src/builtin/system_module.rs [hunk 1]
```diff
@@ -1638,6 +1638,34 @@ register_file_member! {
         false,
         None,
     )
+    mkdir => Type::function(
+        None,
+        Type::str_ref(),
```
Comment (Peefy) [inline kclvm/sema/src/builtin/system_module.rs:1643]: Any type?

File: kclvm/sema/src/builtin/system_module.rs [hunk 2]
```diff
@@ -1638,6 +1638,34 @@ register_file_member! {
...
+                has_default: false,
+            },
+        ],
+        r#"Create a new directory at the specified path if it doesn't already exist."#,
+        false,
+        None,
+    )
+    delete => Type::function(
+        None,
+        Type::str_ref(),
```
Comment (Peefy) [inline kclvm/sema/src/builtin/system_module.rs:1657]: Any type?

File: test/grammar/builtins/file/delete/stdout.golden [hunk 3]
```diff
@@ -0,0 +1 @@
+delete: 'failed to delete 'test_dir': No such file or directory'
```
Comment (Peefy) [inline test/grammar/builtins/file/delete/stdout.golden:1]: You can add the stderr test suite.

**Comment by shruti2522** [comment]
made the changes @Peefy

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**